handler/src/main/java/io/netty/handler/ssl/JdkAlpnApplicationProtocolNegotiator.java

```diff
@@ -1,136 +0,0 @@
-/*
- * Copyright 2014 The Netty Project
- *
- * The Netty Project licenses this file to you under the Apache License,
- * version 2.0 (the "License"); you may not use this file except in compliance
- * with the License. You may obtain a copy of the License at:
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations
- * under the License.
- */
-package io.netty.handler.ssl;
-
-import io.netty.buffer.ByteBufAllocator;
-import javax.net.ssl.SSLEngine;
-
-/**
- * The {@link JdkApplicationProtocolNegotiator} to use if you need ALPN and are using {@link SslProvider#JDK}.
- */
-public final class JdkAlpnApplicationProtocolNegotiator extends JdkBaseApplicationProtocolNegotiator {
```

Comment:
Could you clarify Netty's compatibility contract? I ask because commit 4448b8f42f599e79db1744cd8f5fdfee702c195e recently changed the type signature of `SslEngineWrapperFactory#wrapSslEngine` in a non-backwards-compatible way, despite the fact that that is technically a public method declared on a public interface. Similarly, I judged these classes to be internal Netty classes, as they are not exposed to end users through `ApplicationProtocolConfig` and are, like `wrapSslEngine`, just an implementation detail of Netty's ALPN support.